app/src/colorpalettewidget.cpp

```diff
@@ -71,20 +72,28 @@ void ColorPaletteWidget::initUI()
                              "QPushButton:pressed { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }"
                              "QPushButton:checked { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }";
 
+
+    // color number need to be set to add new color to palette
+    // in cases where the last color was saved
+    // otherwise first color will be black.
+    editor()->color()->setColorNumber(0);
+    editor()->color()->setColor(editor()->color()->frontColor());
```

Comment:
These two lines should probably not be here. Usually in the initialization state, all the current states like the current frame, color, layer will be initialized in the editor & managers, UIs can only pull the states and updates itself. Not to change the states.

Reply:
Hmm it was necessary in the beginning, because otherwise it wouldn't save the previous color, except when you clicked on one of the swatches in the palette. It doesn't seem to be the case anymore though... so I'm removing it